tests/lib/parser.test.ts

```diff
@@ -1,10 +1,14 @@
-import { z } from 'zod/v3';
+import { z as z4 } from 'zod/v4';
+import { z as z3 } from 'zod/v3';
 import { zodResponseFormat } from 'openai/helpers/zod';
 import { makeSnapshotRequest } from '../utils/mock-snapshots';
 
 jest.setTimeout(1000 * 30);
 
-describe('.parse()', () => {
+describe.each([
+  { version: 'v4', z: z4 as any },
+  { version: 'v3', z: z3 as any },
```

Comment:
the `as any` cast isn't ideal, I assume you did that as the types were too complicated and caused errors? if that's the case, then I think we should just cast to `z3` so it's not a complicated union type
```suggestion
  { version: 'v4', z: z4 as typeof z3 },
  { version: 'v3', z: z3 },
```

Reply:
then you could remove the `// @ts-ignore`s further down and just use z3.infer instead of z.infer